nwb_conversion_tools/datainterfaces/ecephys/blackrock/blackrockdatainterface.py

```diff
@@ -33,9 +33,12 @@ def get_source_schema(cls):
         source_schema['properties']['filename']['description'] = 'Path to Blackrock file.'
         return source_schema
     
-    def __init__(self, filename: PathType):
-        nsx_to_load = int(str(filename).split('.')[-1][-1])
-        super().__init__(filename=filename, nsx_to_load=nsx_to_load)
+    def __init__(self, filename: PathType, nsx_override: PathType=None):
+        super().__init__(filename=filename,nsx_override=nsx_override)
+        if self.source_data['nsx_override'] is not None:
+            self.data_filename = self.source_data['nsx_override']
```

Comment:
why not just pass `None`?

Reply:
the method get_metadata_Schema() requires to check for the filename of the nsx file to know the version: ns2/3/4/5/6.  In the case when we need only to open the nsx file and not the nev and other header files (as was in my special case which caused me to propose this change), the path needs to be passed only as under the nsx_override argument : the arguments to the `BlackrockIO` have to be :
```python
bk = BlackrockIO(filename='',nsx_override='filepath-to-nsx.ns3')#would be called by spikeextractors internally. 
```